src/connectionController.ts

```diff
@@ -202,10 +202,22 @@ export default class ConnectionController {
       )
     ).filter((conn): conn is FailedMigrationConnectionDescriptor => !!conn);
 
-    if (Object.keys(this._connections).length) {
+    const loadedConnections = Object.values(this._connections);
+    if (loadedConnections.length) {
       this.eventEmitter.emit(DataServiceEventTypes.CONNECTIONS_DID_CHANGE);
     }
 
+    this._telemetryService.track(TelemetryEventTypes.SAVED_CONNECTIONS_LOADED, {
```

Comment:
We actually don't use the `track()` function from outside of the TelemetryService. We create a helper function for each event we want to track, eg `telemetryService.trackLinkClicked('helpPanel', helpItem.linkId)`. Then we mock this particular event in tests. Maybe the errors you see are because sinon got lost with what is mocked.